docs/sources/flow/setup/deploy-agent.md

```diff
@@ -0,0 +1,125 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/setup/deploy-agent/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/setup/deploy-agent/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/setup/deploy-agent/
+canonical: https://grafana.com/docs/agent/latest/flow/setup/start-agent/
+description: Learn about possible deployment topologies for Grafana Agent
+menuTitle: Deploy Grafana Agent
+title: Grafana Agent deployment topologies
+weight: 900
+---
+
+# Deploying Grafana Agent
+
+Grafana Agent is a flexible, vendor-neutral telemetry collector. This
+flexibility means that the Agent doesn’t enforce a specific deployment topology
+but can work in multiple scenarios.
+
+This page lists common topologies used for deployments of Grafana Agent, when
+to consider using each, issues that users may run into, as well as scaling
+considerations.
+
+## As a centralized collection service
+Deploying the Grafana Agent as a centralized service is recommended for
+collecting application telemetry. This allows a smaller set of agents to
+coordinate service discovery, collection, and remote writing.
+
+![](../../../assets/deploy-agent/centralized-collection.png)
+
+Using this topology requires deploying the Agent on separate infrastructure,
+and making sure that agents can discover and reach these applications over the
+network. The main predictor for the size of the agent is the number of active
+metrics series it is scraping; a rule of thumb is ~10kb of memory for each
+series. We recommend you start looking towards horizontal scaling around the 1M
+active series mark.
+
+### Using Kubernetes StatefulSets
+Deploying Grafana Agent as a StatefulSet is the recommended option for metrics
+collection.
+The persistent pod identifiers make it possible to consistently match volumes
+with pods so that you can use them for the WAL directory.
```

Comment:
We all often talk about WAL, but it may be unclear to newcomers, so perhaps we can add a link to https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.remote_write/ and to Loki's equivalent? Or we could explain it here in one sentence.

Reply:
> If DaemonSets are good enough for logs, not sure why they wouldn't be good for metrics, at least as far as persistent storage is concerned?

Around persistent storage, the idea is that StatefulSets can mount Volumes of different [Storage Classes](https://kubernetes.io/docs/concepts/storage/storage-classes/), eg. SSDs that may be required to keep up with a higher rate of metrics, whereas for logs, the positions file is updates much less frequently and is much smaller than the WAL segments, so there's no issue using the pod's own storage.

Reply:
Interesting, thank you, I didn't know that.